python/lsst/ts/salobj/async_s3_bucket.py

```diff
@@ -0,0 +1,130 @@
+# This file is part of ts_salobj.
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["AsyncS3Bucket"]
+
+import asyncio
+import io
+
+import boto3
+import botocore
+
+
+class AsyncS3Bucket:
+    """Asynchronous interface to Amazon Web Services s3 buckets.
+
+    Parameters
+    ----------
+    name : `str`
+        Name of bucket.
```

Comment:
Could you be a little more specific about what this "name" is? Is it a URI or URL? Or some internal S3 nomenclature?

Reply:
Good suggestion. I added a link to the relevant Amazon Web Services documentation.